src/main/java/org/folio/inventory/dataimport/handlers/matching/loaders/ItemLoader.java

```diff
@@ -62,7 +63,8 @@ protected String addCqlSubMatchCondition(DataImportEventPayload eventPayload) {
         JsonObject itemAsJson = new JsonObject(eventPayload.getContext().get(EntityType.ITEM.value()));
         cqlSubMatch = format(" AND id == \"%s\"", itemAsJson.getString("id"));
       } else if (isNotEmpty(eventPayload.getContext().get(EntityType.HOLDINGS.value()))) {
-        JsonObject holdingAsJson = new JsonObject(eventPayload.getContext().get(EntityType.HOLDINGS.value()));
+        JsonObject holdingAsJson = new JsonArray(eventPayload.getContext().get(EntityType.HOLDINGS.value()))
+          .getJsonObject(0);
```

Comment:
@yaroslav-epam  In current karate test failure, this line errors sometime with a index out of range exception. That is preventing DI_INVENTORY_ITEM_NOT_MATCHED to be published by mod-inventory to allow a discarded message to show in journal logs.

we should change this to account for empty array scenarios.

Reply:
The exception is not being logged as well so we make modifications to allow proper exception propagation.